### DISCUSSION TITLE
willmore energy

### DISCUSSION THREAD

**Discussion by rob-rb** [meshpro/meshplex#163] [General]
Hi Nico,

I'd like to calculate the intersection angles of the circumcircles.  [https://page.math.tu-berlin.de/~bobenko/papers/2005_Bob.pdf]
This operation is a per edge operation. But I need for each edge the 2 corresponding cells. Could you give maybe a hint howto get from edges to cells?

**Comment by nschloe** [comment]
face->cells relations is the only upstream relation I implemented, so you're in luck. :smile: Check `mesh.facets_cells` and https://github.com/nschloe/meshplex/blob/main/src/meshplex/_mesh.py#L780.
